### DISCUSSION TITLE
Variants (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4809] [Show and tell]
Motoko has a neat feature where a datatype can have exactly one choice of many, and can also carry specific data. These are called _variants_ and generalise other languages' _enumerations_.
``` Motoko
type Certainty<T> = { #yes : T; #maybe (Float, T); #no };
```
You can (usually) figure which case you have at hand by performing a _pattern match_
``` Motoko
example TBW
```

**Comment by jessiemongeon1** [comment]
Scheduled for Friday 4/18
